### PR TITLE
Demo verifier state continued

### PR DESCRIPTION
**Pull request by xqft**
This adds:
1. a bash script for querying data from the last block of the best tip
2. a Rust binary for taking that data and serializing it in MessagePack
3. a deserializer function for this specific data which will be stored in the contract if its associated proof is correct
4. instrucions inside `eth_verifier/README.md` for how to deploy and run the contract

### CODE REVIEW

File: demo/eth_verifier/verify.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+export CONTRACT_ADDR=0x0 # replace this after deploying
```
Comment (pablodeymo) [inline demo/eth_verifier/verify.sh:3]: we should receive this value as an argument in the invocation

File: demo/eth_verifier/state_serializer
Comment (pablodeymo) [inline demo/eth_verifier/state_serializer:1]: I think we don't need this file commited in the repo.

File: demo/eth_verifier/Makefile [hunk 2]
```diff
@@ -17,6 +17,17 @@ deploy_verifier:
 	PRIVATE_KEY=0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80 \
 	forge script script/Deploy.s.sol:Deploy --via-ir --broadcast --rpc-url http://localhost:8545
 
-run_verifier:
+query:
+	./query.sh | xargs -0 ./state_serializer > state.mpk
```
Comment (pablodeymo) [inline demo/eth_verifier/Makefile:21]: the state_serializer file should be compiled in the Makefile as part of the execution

File: demo/eth_verifier/verify.sh (hunk 1 above)
Comment (xqft) [reply demo/eth_verifier/verify.sh]: done!

File: demo/eth_verifier/state_serializer
Comment (xqft) [reply demo/eth_verifier/state_serializer:1]: done!

File: demo/eth_verifier/Makefile (hunk 2 above)
Comment (xqft) [reply demo/eth_verifier/Makefile]: done!

**Review by pablodeymo — APPROVED** [review]
LGTM

**Pull request merged**